packages/block-library/package.json

```diff
@@ -30,6 +30,13 @@
 		"src/**/*.scss",
 		"{src,build,build-module}/*/init.js"
 	],
+	"wp-script-module-exports": {
+		"./file/view": "./build-module/file/view.js",
```

Comment:
It starts to look like `exports` from `package.json`. Nice! 👍🏻

Reply:
Yes! I think this is going to be a very good direction. It moves in the direction of exports, but doesn't switch all the packages immediately.

It's also a custom field, and these are custom builds in general specific for use as WordPress Script Modules. It makes it clear that they're not intended for general consumption.

Reply:
I also think we can move in the direction of generating the entrypoints in a standard way by reading package.json files.

Reply:
Unfortunately, it doesn't work 🙃  (I thought it did but I'm not sure why that was).

Problems:
When entrypoints are paths (`./packages/block-library/file/view`), webpack never sees the package.json in order to find these paths.

>   ERROR in block-library/file/view
>  Module not found: Error: Can't resolve './packages/block-library/file/view' in '…'

When entrypoints are to modules (`@wordpress/block-library/file/view`) webpack resolves these correctly, but they're handled as externals. DEWP rejects these. We could fix DEWP, but even then we have entrypoints like `@wordpress/interactivity` which DEWP will externalize and there doesn't seem to be a good way to say "Externalize this _except_ when it's the entrypoint." Externalizing entrypoints is, needless to say, bad and broken 🙂 

So I don't think actually using this field as webpack `exportsFields` is going to work, but we can read this field and build a path to it programatically.

Reply:
> So I don't think actually using this field as webpack exportsFields is going to work, but we can read this field and build a path to it programatically.

That sounds like a great first step with some thin compact layer in the webpack config that scans package.json files and produces entry points that are hardcoded today. For blocks you could eventually use pattern matching.